kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaServiceReconciler.java

```diff
@@ -133,28 +135,32 @@ public UpdateControl<KafkaService> reconcile(KafkaService service, Context<Kafka
 
         KafkaService updatedService = null;
 
+        Optional<ConfigMap> maybeTrustAnchorConfigMap = context.getSecondaryResource(ConfigMap.class, CONFIG_MAPS_EVENT_SOURCE_NAME);
         var trustAnchorRefOpt = Optional.ofNullable(service.getSpec())
                 .map(KafkaServiceSpec::getTls)
                 .map(Tls::getTrustAnchorRef);
         if (trustAnchorRefOpt.isPresent()) {
-            updatedService = checkTrustAnchorRef(service, context, trustAnchorRefOpt.get());
+            updatedService = checkTrustAnchorRef(service, trustAnchorRefOpt.get(), maybeTrustAnchorConfigMap);
         }
 
+        Optional<Secret> maybeSecret = context.getSecondaryResource(Secret.class, SECRETS_EVENT_SOURCE_NAME);
         if (updatedService == null) {
             var certRefOpt = Optional.ofNullable(service.getSpec())
                     .map(KafkaServiceSpec::getTls)
                     .map(Tls::getCertificateRef);
             if (certRefOpt.isPresent()) {
                 String path = "spec.tls.certificateRef";
-                updatedService = ResourcesUtil.checkCertRef(service, context, SECRETS_EVENT_SOURCE_NAME, certRefOpt.get(), path, statusFactory);
+                updatedService = ResourcesUtil.checkCertRef(service, certRefOpt.get(), path, statusFactory, maybeSecret);
             }
         }
 
         if (updatedService == null) {
-            updatedService = statusFactory.newTrueConditionStatusPatch(service, ResolvedRefs);
+            HasMetadata[] hasMetadata = Stream.concat(maybeTrustAnchorConfigMap.stream(), maybeSecret.stream()).toArray(HasMetadata[]::new);
+            String checksum = hasMetadata.length > 0 ? MetadataChecksumGenerator.checksumFor(hasMetadata) : MetadataChecksumGenerator.NO_CHECKSUM_SPECIFIED;
```

Comment:
This zero length semantic looks to be common with the other usages. Is there any reason why we don't push the logic into `checksumFor()`?